packages/vite/src/node/config.ts

```diff
@@ -477,12 +477,34 @@ export async function resolveConfig(
   const resolvedRoot = normalizePath(
     config.root ? path.resolve(config.root) : process.cwd(),
   )
+
+  // We will warn if the root path includes either the '#' or '?' character
+  const badPathCharacters = []
+
   if (resolvedRoot.includes('#')) {
+    badPathCharacters.push('#')
+  }
+
+  if (resolvedRoot.includes('?')) {
+    badPathCharacters.push('?')
+  }
+
+  if (badPathCharacters.length > 0) {
+    const characterString =
+      badPathCharacters.length > 1
+        ? badPathCharacters.map((c) => `"${c}"`).join(' and ')
+        : badPathCharacters.map((c) => `"${c}"`).join('')
```

Comment:
I think we can directly use `badPathCharacters.map((c) => `"${c}"`).join(' and ')` here as single element arrays won't use the `and`.

Reply:
Lol, of course! I don't know what I was thinking 😆